src/client/options.rs

```diff
@@ -45,7 +44,10 @@ use crate::{
     srv::{OriginalSrvInfo, SrvResolver},
 };
 
+#[cfg(feature = "dns-resolver")]
 pub use resolver_config::ResolverConfig;
+#[cfg(not(feature = "dns-resolver"))]
+pub(crate) use resolver_config::ResolverConfig;
```

Comment:
Use of DNS resolution is ... not neatly encapsulated in the codebase, so to keep this from being a giant refactor PR I made the various machinery crate-internal when the feature is disabled rather than being feature-gated entirely.